index.d.ts

```diff
@@ -0,0 +1,525 @@
+/**
+ * Access the Electron APIs in both the main and renderer process without having to care about which one you're in.
+ *
+ * @example
+ *
+ * // In the renderer process:
+ * api.app.quit(); // The `app` API is usually only available in the main process.
+ */
+export const api: Electron.AllElectron | Electron.Remote;
+
+/**
+ * Check for various things.
+ */
+export const is: {
+	/**
+	 * Running on macOS.
+	 */
+	macos: boolean,
+
+	/**
+	 * Running on Linux.
+	 */
+	linux: boolean,
+
+	/**
+	 * Running on Windows.
+	 */
+	windows: boolean,
+
+	/**
+	 * Running on the [main process](https://electronjs.org/docs/tutorial/quick-start/#main-process).
+	 */
+	main: boolean,
+
+	/**
+	 * Running on the [renderer process](https://electronjs.org/docs/tutorial/quick-start/#renderer-process).
+	 */
+	renderer: boolean,
+
+	/**
+	 * The app is using [ASAR](https://electronjs.org/docs/tutorial/application-packaging/).
+	 */
+	usingAsar: boolean,
+
+	/**
+	 * Running in development, not in production.
+	 */
+	development: boolean,
+
+	/**
+	 * The app is an Mac App Store build.
+	 */
+	macAppStore: boolean,
+
+	/**
+	 * The app is a Windows Store AppX build.
+	 */
+	windowsStore: boolean
+};
+
+/**
+ * Electron version.
+ *
+ * @example
+ *
+ * '1.7.9'
+ */
+export const electronVersion: string;
+
+/**
+ * Chrome version in Electron.
+ *
+ * @example
+ *
+ * '62.0.3202'
+ */
+export const chromeVersion: string;
+
+type RequireAtLeastOne<T, Keys extends keyof T = keyof T> =
+	Pick<T, Exclude<keyof T, Keys>>
+	& {
+		[K in Keys]-?: Required<Pick<T, K>>
+	}[Keys]
+
+interface _Choices<M, W, L, D> {
```

Comment:
Use descriptive type names if more than one, for readability.

`M` => `Macos`
`W` => `Windows`
etc